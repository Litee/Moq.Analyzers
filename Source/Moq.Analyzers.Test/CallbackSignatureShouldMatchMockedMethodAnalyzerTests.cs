using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class CallbackSignatureShouldMatchMockedMethodAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldPassIfGoodParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/CallbackSignatureShouldMatchMockedMethod.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new CallbackSignatureShouldMatchMockedMethodAnalyzer();
        }
    }
}