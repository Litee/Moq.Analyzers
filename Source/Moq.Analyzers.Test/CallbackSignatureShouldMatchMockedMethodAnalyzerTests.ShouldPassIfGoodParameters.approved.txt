Diagnostic 1
	Id: Moq1100
	Location: SourceFile(Test0.cs[445..452))
	Highlight: (int i)
	Lines: mock.Setup(x => x.Do(It.IsAny<string>())).Callback((int i) => { });
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 2
	Id: Moq1100
	Location: SourceFile(Test0.cs[526..548))
	Highlight: (string s1, string s2)
	Lines: mock.Setup(x => x.Do(It.IsAny<string>())).Callback((string s1, string s2) => { });
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 3
	Id: Moq1100
	Location: SourceFile(Test0.cs[661..680))
	Highlight: (string s1, int i1)
	Lines: mock.Setup(x => x.Do(It.IsAny<int>(), It.IsAny<string>(), It.IsAny<DateTime>())).Callback((string s1, int i1) => { });
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 4
	Id: Moq1100
	Location: SourceFile(Test0.cs[760..767))
	Highlight: (int i)
	Lines: mock.Setup(x => x.Do(It.IsAny<List<string>>())).Callback((int i) => { });
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

